### ISSUE TITLE
Triggers?

### ISSUE THREAD

**Issue by drussellmrichie** [jakdot/pyactr#5]
Hi, great book and package so far!!! I've been interested in ACT-R for a while but haven't found anything else nearly as accessible as this! Can't wait to get to the rest of it! 😄 

I'm working on the exercise to extend the lexical decision model to respond to a sequence of stimuli, and besides changing the production rules, one thing I have to do is set up triggers so that button presses cause the next word to appear. I thought maybe I could do something like this:

```
lex_dec_sim = lex_decision.simulation(
    realtime=False,
    gui=False,
    environment_process=environment.environment_process,
    stimuli=word,
    triggers=['J','F'],
    times=1)
```

That is, I thought I could pass a list of key presses that should present the next stimulus. But this doesn't seem to work? Is there documentation somewhere on triggers?

**Comment by jakdot** [comment]
Hi, Great to hear that you find the package and the book useful.

Your code for the triggers is correct but you also need to modify stimuli. Right now (if you took over the code from the book), there is just one stimulus (one word). You would need to create a list of stimuli.

`words = [{1: {'text': 'elephant', 'position': (320, 180)}},
         {1: {'text': 'not_a_word', 'position': (320, 180)}}]
lex_dec_sim = lex_decision.simulation(
    realtime=False,
    gui=False,
    environment_process=environment.environment_process,
    stimuli=words,
    triggers=['J','F'],
    times=1)
`

**Comment by drussellmrichie** [comment]
Thanks for the reply. Yes, I did do that. Actually I included three words in the list of stimuli. Here is my full code:

https://gist.github.com/drussellmrichie/c76fcb860c8c9e00d85647d1a34be14e

At the bottom I mention that this produces the error `ACTRError: In environment, stimuli must be the same length as triggers or one of the two must be of length 1`

I then also tried `triggers=[('J','F')]*3`, which gave the error `ACTRError: Triggers are not strings; currently nothing else than strings are allowed as triggers in environment`

**Comment by jakdot** [comment]
The issue in that code is that there are three stimuli (elephant, wug, dog), but only two triggers (J, F). You have to add a third trigger, so triggers and stimuli are of equal length. So triggers=['J', 'F', 'J'].

**Comment by drussellmrichie** [comment]
I see. So you weren't imagining that we'd allow incorrect responses for the
purposes of this exercise?

**Comment by jakdot** [comment]
No, we did not assume that. But you are right, that would be one way to approach the question and it would be very useful to model correct and incorrect responses as triggers, not just for the purpose of this exercise, but in general. I will add that functionality.

**Comment by jakdot** [comment]
I updated pyactr. You can now specify multiple triggers per screen. You do that either by putting all the triggers in one string, say, triggers='jf' would trigger the screen when j or f is pressed, or you can put all possible triggers inside a list of iterables, say, triggers=[('j', 'f')]. Note that if you just put them in a list, say, triggers=['j', 'f'], this has a different meaning - it means that the first screen is triggered by 'j' and the second screen is triggered by 'f'.

**Comment by drussellmrichie** [comment]
That's excellent. Thank you so much!
